src/v/hashing/crc32.h

```diff
@@ -11,7 +11,7 @@
 
 #pragma once
 #include <cstdint>
-#include <zlib.h>
+#include <zlib.h> // provides ::crc32
```

Comment:
I'm not sure I follow why this causes "accidental and unnecessary recompilation", and why adding it back will prevent that?

Reply:
because I was removing unused headers, and by all accounts, zlib didn't provide the crc32 function. but discovering that invalidated all of my build targets.